ruma-client-api/src/r0/capabilities/get_capabilities.rs

```diff
@@ -114,6 +118,20 @@ impl Capabilities {
 
         Ok(())
     }
+
+    /// Returns an iterator over the capabilities
+    pub fn iter(&self) -> CapabilitiesIter {
```

Comment:
I'm surprised this works, considering the `CapabilitiesIter` is a private type, but this function is public. I'm pretty sure the `iter` mod needs to be made public so `CapabilitiesIter` and maybe more importantly `CapabilityRef` (its methods) can actually be used.

Reply:
Do you mean that these types cannot be used outside of `get_capabilities.rs`, as the iter mod is private?
I tried writing `capabilities_iter` test and using those in `r0/capabilities.rs` and it worked fine. So I am not sure if we need to do that?

Reply:
Well the test is still inside the crate. Try whether it still works when moving it into a file in the tests directory, I'd expect it to fail to compile then.

Reply:
I am not sure if I am doing the right thing, but i wrote this test in `ruma/ruma/tests` directory
```rust
#[cfg(test)]
mod tests {
    #[test]
    fn capabilities_iter() {
        assert!(cfg!(feature = "ruma-client-api"));
        use ruma_client_api::r0::capabilities::get_capabilities::Capabilities;

        let caps = Capabilities::new();
        let mut caps_iter = caps.iter();

        let iter_res = caps_iter.next().unwrap();
        assert_eq!(iter_res.name(), "m.change_password");

        let iter_res = caps_iter.next().unwrap();
        assert_eq!(iter_res.name(), "m.room_versions");

        assert!(caps_iter.next().is_none());
    }
}
```
ran it with `cargo test --features=ruma-client-api` and it works fine

Reply:
I'll look into what's happening here later...

Reply:
Okay, so apparently Rust doesn't consider the types in `iter` private even though they can't be named from outside the crate. I think having `CapabilityRef` be an unnamable type whose methods can be used is not super nice though, so could you make the `iter` module public and apply the other visibility suggestion from below?

Also `Capabilities` + the iter module should maybe move one level up, i.e. be at `r0::capabilities::{iter, Capabilities}` instead of `r0::capabilities::get_capabilities::{iter, Capabilities}`. But you don't need to do that in this PR, I can also do it afterwards.